### ISSUE TITLE
Download URL.

### ISSUE THREAD

**Issue by rnaby** [YahnisElsts/plugin-update-checker#59]
First of all your work is awesome. Now I've two question,

1. JSON file. Right now your update checker use json file. Is there any way to make it use the json response ???

2. Right now the plugin update checker needs a "download_url" as a must. But I need to make like if anyone click on it it will redirect to my site where customer can download and manually install it. If this is not possible, could it be done that there will be no install button for my plugin. Just show the message that it has update. So client can came to my site and download the update manually ???

Thanks in advance.

**Comment by YahnisElsts** [comment]
> JSON file. Right now your update checker use json file. Is there any way to make it use the json response ???

Sorry, can you rephrase that? I'm not sure what you mean. 

The simplest way to provide update information is to create a JSON file, but you could also write a script that outputs some JSON and then point the update checker at that script. It would work equally well.

> Right now the plugin update checker needs a "download_url" as a must.

That's not entirely right. In practice, you *can* omit the `download_url` field. This will cause WordPress to display a message like "automatic updates are not available for this plugin" on the "Plugins" page. 

> But I need to make like if anyone click on it it will redirect to my site where customer can download and manually install it.

You'll have to implement that yourself. There's no built-in feature like that.

**Comment by rnaby** [comment]
Thanks man for replying so quick. :) 


Let me clear the first question. 

`require 'plugin-update-checker/plugin-update-checker.php';
$myUpdateChecker = PucFactory::buildUpdateChecker(
    'http://example.com/path/to/metadata.json',
    __FILE__
);`

Here look, I need to provide a JSON file

`'http://example.com/path/to/metadata.json',`

I need to make it like 
 
`'http://example.com/path/to/',`

no JSON file. Only JSON response. So your above solution for this is not fully clear to me. If you give me any solution on this context, it would be so helpful.

Thanks again.

**Comment by YahnisElsts** [comment]
Err, did you mean to close this? I saw an email notification about a new comment from you, but it doesn't appear to be here.

**Comment by rnaby** [comment]
I just got it man. :) Thanks though for your reply. :+1: 

By the way my question was if I make the JSON URL like 
`'http://example.com/path/to/'`
instead of this
`'http://example.com/path/to/metadata.json'`
would it work ? I think it's gonna work, isn't it ?

**Comment by YahnisElsts** [comment]
It depends on what `http://example.com/path/to/` returns. If you get a JSON document when you visit that URL then sure, it should work as a metadata URL for the update checker.
